cmd/branchingconfigmanagers/bugzilla-config-manager/main.go

```diff
@@ -0,0 +1,197 @@
+		if ocpProduct != "ocp" {
+			continue
+		}
+
+		for productVersion, config := range productConfig {
+			if len(config.Previous) < 1 {
+				continue
+			}
+
+			// TODO: We assume the first event is the last - Is that reasonable?
+			switch config.Previous[0].Event {
+			case ocplifecycle.LifecycleEventGenerallyAvailable:
+				parsedVersion, err := ocplifecycle.ParseMajorMinor(productVersion)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse %s as majorMinor version: %w", productVersion, err))
+					continue
+				}
+				gaVersions = append(gaVersions, *parsedVersion)
+			case ocplifecycle.LifecycleEventOpen:
+				if developmentVersionFound {
+					errs = append(errs, fmt.Errorf("found multiple development versions: %s and %s", developmentVersion, productVersion))
+				}
+				parsedVersion, err := ocplifecycle.ParseMajorMinor(productVersion)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse %s as majorMinor version: %w", productVersion, err))
+					continue
+				}
+				developmentVersion = parsedVersion
+				developmentVersionFound = true
+			}
+		}
+	}
+
+	return developmentVersion, gaVersions, utilerrors.NewAggregate(errs)
+}
+
+func reconcileConfig(cfg *plugins.Bugzilla, developmentVersion *ocplifecycle.MajorMinor, gaVersions []ocplifecycle.MajorMinor) *plugins.Bugzilla {
+	if cfg.Default == nil {
+		cfg.Default = map[string]plugins.BugzillaBranchOptions{}
+	}
+
+	if developmentVersion != nil {
+		// main/master: Set developmentVersion and leave the rest as is
+		for _, developmentBranchName := range []string{"main", "master"} {
+			config := cfg.Default[developmentBranchName]
+			config.TargetRelease = utilpointer.String(developmentVersion.String() + ".0")
+
+			cfg.Default[developmentBranchName] = config
+		}
+
+		// developmentVersion/developmentVersion+1: Set config for {release/openshift}-$VERSION:
+		// * dependentBugStatus: ["MODIFIED", "ON_QA", "VERIFIED"]
+		// * DependentBugTargetReleases: "$VERSION+1.0"
+		// * ValidateByDefault: true
+		for _, developmentVersion := range []ocplifecycle.MajorMinor{*developmentVersion, developmentVersion.WithIncrementedMinor(1)} {
+			for _, releaseBranchPrefix := range releaseBranchPrefixes {
+				cfg.Default[releaseBranchPrefix+developmentVersion.String()] = plugins.BugzillaBranchOptions{
+					DependentBugStates:         developmentDependentBugStates,
+					DependentBugTargetReleases: &[]string{developmentVersion.WithIncrementedMinor(1).String() + ".0"},
+					TargetRelease:              utilpointer.String(developmentVersion.String() + ".0"),
+					ValidateByDefault:          utilpointer.Bool(true),
+				}
+			}
+		}
+	}
+
+	// GA versions:
+	// * DependentBugTargetReleases: "$VERSION+1.0"
+	// * if not latestGA: add $Version+1.z to DependentBugTargetReleases
+	// * TargetRelease: $VERSION.z
+	// * ValidateByDefault: true
+	sort.Slice(gaVersions, func(i, j int) bool { return gaVersions[i].String() < gaVersions[j].String() })
```

Comment:
Does `4.11` sort correctly w.r.t `4.1` ?

Reply:
Yeah seems so, I've added an explicit testcase for that, the generated `/testdata/zz_fixture_TestRun_lifecycle_multiple_ga_411_biger_41.yaml.yaml` has the `z` `dependent_bug_target_release` for `4.1` and not `4.11`.